dev/TeachingTip/TeachingTip.cpp

```diff
@@ -916,6 +946,10 @@ void TeachingTip::ClosePopup()
     {
         m_popup.get().IsOpen(false);
     }
+    if (m_lightDismissIndicatorPopup)
```

Comment:
This kind of pattern would be more efficient (and a little more readable) as:
```c++
if (auto lightDismissIndicatorPopup = m_lightDismissIndicatorPopup.get())
{
    lightDismissIndicatorPopup.IsOpen(false);
}
```

Can you start adopting this in this codebase? (no need to retrofit but be on the lookout for it when changing code).

Reply:
This is slightly less efficient in the scenarios where m_lightDismissIndicatorPopup is null, correct? We don't expect this to happen very often so I still think this is a good suggestion.

Reply:
You mean my proposed change is less efficient? Because there's a temporary variable that the compiler doesn't know it can elide the destructor for?

Reply:
Yes, I suspect the proposed change is less efficient. In this scenario all you've done is moved the getter to the if check so this changed doesn't reduce the number of QI's just pushes it somewhere else, and if the pointer is to null we actually do an extra QI right?

Reply:
Let me compile two functions with the proposed code in question and we can discuss what the disasm looks like. Stay tuned. :)

Reply:
I figured this out. So my recommendation needs to be:
`if (auto&& lightDismissIndicatorPopup = m_lightDismissIndicatorPopup.get())` OR `if (const auto& lightDismissIndicatorPopup = m_lightDismissIndicatorPopup.get())`. Simply `auto` will do a by-value copy into a local which needs an addref/release.

But in investigating this I also discovered that the return type of the tracker_ref.get() function was returning by value instead of by reference. After fixing that then the codegen for both approaches is the same. That means that now the choice between the two is merely stylistic.

Related fix is #341